ffi/input_pocketbook.lua

```diff
@@ -26,23 +32,32 @@ local function genEmuEvent(t,c,v)
     })
 end
 
-local nTouch = 0
+-- EV_KEY
+local raw_keymap
+local poll_fds, poll_fds_count = nil, 0
+local num_touch = 0
+
+local function getTimestamp()
+    local sec, usec = util.gettime()
+    ts = { sec = sec, usec = usec }
+    return ts
+end
+
 local function translateEvent(t, par1, par2)
     if eventq == nil then
         return 0
     end
-    local sec, usec = util.gettime()
-    ts = { sec = sec, usec = usec }
+    getTimestamp()
```

Comment:
Stupid question: isn't there a scope issue here?

(I genuinely have no idea how local functions behave in Lua: is `getTimestamp`'s scope tighter than `translateEvent`'s, or does this behave like a C macro, so that `translateEvent` now has access to `getTimestamp`'s `sec` & `usec` variables?

Reply:
Nope, `ts` is simply a "global" (as in, in the outermost scope) `gts()` just updates it with current value, as we're soon to be generating events with same timestamp in bulk (so they all share same ts). Later on, genEmuEvent picks it up from the same outer scope. I suppose this could be renamed to updateTimestamp() to make it more clear.

Reply:
Oh, duh, I missed the scoping the `ts`, was too focused on `sec` & `usec`, thanks ;)